supervision/metrics/core.py

```diff
@@ -0,0 +1,210 @@
+                self._data_1[class_id] = np.vstack(
+                    (self._data_1[class_id], content_of_class)
+                )
+
+        if content_2 is not None and len(content_2) > 0:
+            assert len(content_2) == len(class_ids_2)
+            for class_id in set(class_ids_2):
+                content_of_class = content_2[class_ids_2 == class_id]
+                if class_id not in self._data_2:
+                    self._data_2[class_id] = content_of_class
+                    continue
+                self._data_2[class_id] = np.vstack(
+                    (self._data_2[class_id], content_of_class)
+                )
+
+    def __iter__(
+        self,
+    ) -> Iterator[Tuple[int, Optional[npt.NDArray], Optional[npt.NDArray]]]:
+        class_ids = sorted(
+            set.union(set(self._data_1.keys()), set(self._data_2.keys()))
+        )
+        for class_id in class_ids:
+            yield (
+                class_id,
+                self._data_1.get(class_id, None),
+                self._data_2.get(class_id, None),
+            )
+
+    def _get_content(
+        self, data: Union[npt.NDArray, Detections]
+    ) -> Optional[npt.NDArray]:
+        """Return boxes, masks or oriented bounding boxes from the data."""
+        if not isinstance(data, (Detections, np.ndarray)):
+            raise ValueError(
+                f"Invalid data type: {type(data)}. Only Detections or np.ndarray are supported."
+            )
+        if isinstance(data, np.ndarray):
+            return data
+
+        if self._metric_target == MetricTarget.BOXES:
+            return data.xyxy
+        if self._metric_target == MetricTarget.MASKS:
+            return data.mask
+        if self._metric_target == MetricTarget.ORIENTED_BOUNDING_BOXES:
+            obb = data.data.get(config.ORIENTED_BOX_COORDINATES, None)
+            if isinstance(obb, list):
+                obb = np.array(obb, dtype=np.float32)
+            return obb
+        raise ValueError(f"Invalid metric target: {self._metric_target}")
+
+    def _get_class_ids(
+        self, data: Union[npt.NDArray, Detections]
+    ) -> npt.NDArray[np.int_]:
+        if self._class_agnostic or isinstance(data, np.ndarray):
+            return np.array([CLASS_ID_NONE] * len(data), dtype=int)
+        assert isinstance(data, Detections)
+        if data.class_id is None:
+            return np.array([CLASS_ID_NONE] * len(data), dtype=int)
+        return data.class_id
+
+    def _validate_class_ids(self, class_id: npt.NDArray[np.int_]) -> None:
+        class_set = set(class_id)
+        if len(class_set) >= 2 and -1 in class_set:
+            raise ValueError(
+                "Metrics store received results with partially defined classes."
+            )
+
+    def _validate_shape(self, data: npt.NDArray) -> None:
+        if self._datapoint_shape is None:
+            assert self._metric_target == MetricTarget.MASKS
+            self._datapoint_shape = data.shape[1:]
```

Comment:
What's happening here? What if your dataset contained images of different sizes and, by extent, masks of different sizes?